ipheghe/More-Recipes
# feat(unfavorite): implement unfavorite recipe action

### What does this PR do?
Users should be able to unfavorite already favorited recipe

### Description of Task to be completed?
- add unfavorite recipe controller function
- add unfavorite recipe route
- integrate into client side on view recipe component

### How should this be manually tested?
- clone repo or switch to branch 
- run `git pull` on the branch 
- start server on localhost
- navigate to http://localhost:8000/#!/login
- try entering invalid credentials
- on the dashboard page, click on more button on the any of the recipe card
- it takes the viewRecipe page for the recipe clicked on
- if recipe has not been favorited before by the user, it should show favorite button
- if recipe has been favorited by user before, it should show unfavorite button
- if unfavorite button is clicked, recipe is deleted from user favorite list

### Any background context you want to provide?
N/A

## What are the relevant pivotal tracker stories?
story type: feature
story Id: 152846729
story title: User should be able to unfavorite recipe

### Screenshots (if appropriate)

![image](https://user-images.githubusercontent.com/29812424/32903587-a6a87aa0-caf5-11e7-97df-79bf31d39b2b.png)

![image](https://user-images.githubusercontent.com/29812424/32903597-af125864-caf5-11e7-9e76-3d3922b6e361.png)


### Questions:
N/A
